harness/propertyHelper.js

```diff
@@ -1,13 +1,17 @@
 
 function isConfigurable(obj, name) {
+    var descriptor = Object.getOwnPropertyDescriptor(obj, name);
     try {
         delete obj[name];
     } catch (e) {
         if (!(e instanceof TypeError)) {
             $ERROR("Expected TypeError, got " + e);
         }
     }
-    return !Object.prototype.hasOwnProperty.call(obj, name);
+    var result = !Object.prototype.hasOwnProperty.call(obj, name);
+    // Restore property (no-op if not deleted)
+    Object.defineProperty(obj, name, descriptor);
```

Comment:
while it might not leave side effects beyond Proxies, it's beyond the responsibility for verifyConfigurable restore a state, as an extra step. Tests should follow the minimum possible steps to confirm an assertion. The best of all options would be removing the delete operation, but that's necessary as we already found parts where the descriptor was not matching the actual configurability of a property.

If the property is sent to be tested it should be consumed and that's it.

There's also the problem with non extensible objects:

```js 
Object.preventExtensions(Object);
delete Object.length;
Object.defineProperty(Object, 'length', { value: 42 }); // will throw
```